src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder_ListPatterns.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class DecisionDagBuilder
+    {
+        private void MakeTestsAndBindingsForLengthAndListPatterns(BoundDagTemp input, BoundRecursivePattern recursive, ArrayBuilder<BoundPatternBinding> bindings, ArrayBuilder<Tests> tests)
+        {
+            if (recursive.HasErrors)
+                return;
+
+            Debug.Assert(recursive.LengthProperty is not null || input.Type.IsSZArray());
+            var syntax = recursive.Syntax;
+            var lengthProperty = recursive.LengthProperty ?? (PropertySymbol)_compilation.GetSpecialTypeMember(SpecialMember.System_Array__Length);
```

Comment:
What happens if `Array.Length` is missing?

Reply:
Should we even be worry about *special members*? My understanding is that only *well-known members* can be missing.

Reply:
We do :-/
Two ways to achieve this: defined a tiny corlib in test with `CreateEmptyCompilation` or use `MakeTypeMissing(SpecialType type)`/`MakeMemberMissing(SpecialMember member)`.